### ISSUE TITLE
Documentation out of date for summarization

### ISSUE THREAD

**Issue by silent-vim** [OpenNMT/OpenNMT-py#820]
Hi, when trying to run the summarization example with CNN/DM dataset the training abruptly finishes without any errors. The documentation seems to be working with commit [0397acd](https://github.com/OpenNMT/OpenNMT-py/commit/0397acda33531889dd2d62984b162305d48d9d10#diff-86631deba6b1df4b7aab7c6da70c7c5d) when it was last updated. 

The same command mentioned on the examples page proceeds gradually through training and gives expected results through inference. 

Is there an example of summarization with the recent code changes since commit 0397acd. 

Thank you.

## Logs for latest master branch 
Training finishes within 5 min at the last statement and model is not saved unless you change save_checkpoint flag
 
```
[2018-07-11 06:57:39,737 INFO] Loading train dataset from data/cnndm/CNNDM.train.1.pt, number of examples: 126095
[2018-07-11 06:57:39,737 INFO] Loading train dataset from data/cnndm/CNNDM.train.1.pt, number of examples: 126095
[2018-07-11 06:57:40,103 INFO]  * vocabulary size. source = 50004; target = 50004
[2018-07-11 06:57:40,103 INFO]  * vocabulary size. source = 50004; target = 50004
[2018-07-11 06:57:40,103 INFO] Building model...
[2018-07-11 06:57:40,103 INFO] Building model...
[2018-07-11 06:57:45,213 INFO] NMTModel(
  (encoder): RNNEncoder(
    (embeddings): Embeddings(
      (make_embedding): Sequential(
        (emb_luts): Elementwise(
          (0): Embedding(50004, 128, padding_idx=1)
        )
      )
    )
    (rnn): LSTM(128, 256, bidirectional=True)
    (bridge): ModuleList(
      (0): Linear(in_features=256, out_features=256, bias=True)
      (1): Linear(in_features=256, out_features=256, bias=True)
    )
  )
  (decoder): InputFeedRNNDecoder(
    (embeddings): Embeddings(
      (make_embedding): Sequential(
        (emb_luts): Elementwise(
          (0): Embedding(50004, 128, padding_idx=1)
        )
      )
    )
    (dropout): Dropout(p=0.0)
    (rnn): StackedLSTM(
      (dropout): Dropout(p=0.0)
      (layers): ModuleList(
        (0): LSTMCell(640, 512)
      )
    )
    (attn): GlobalAttention(
      (linear_context): Linear(in_features=512, out_features=512, bias=False)
      (linear_query): Linear(in_features=512, out_features=512, bias=True)
      (v): Linear(in_features=512, out_features=1, bias=False)
      (linear_out): Linear(in_features=1024, out_features=512, bias=True)
      (softmax): Softmax()
      (tanh): Tanh()
    )
  )
  (generator): CopyGenerator(
    (linear): Linear(in_features=512, out_features=50004, bias=True)
    (linear_copy): Linear(in_features=512, out_features=1, bias=True)
    (softmax): Softmax()
    (sigmoid): Sigmoid()
  )
)
[2018-07-11 06:57:45,213 INFO] NMTModel(
  (encoder): RNNEncoder(
    (embeddings): Embeddings(
      (make_embedding): Sequential(
        (emb_luts): Elementwise(
          (0): Embedding(50004, 128, padding_idx=1)
        )
      )
    )
    (rnn): LSTM(128, 256, bidirectional=True)
    (bridge): ModuleList(
      (0): Linear(in_features=256, out_features=256, bias=True)
      (1): Linear(in_features=256, out_features=256, bias=True)
    )
  )
  (decoder): InputFeedRNNDecoder(
    (embeddings): Embeddings(
      (make_embedding): Sequential(
        (emb_luts): Elementwise(
          (0): Embedding(50004, 128, padding_idx=1)
        )
      )
    )
    (dropout): Dropout(p=0.0)
    (rnn): StackedLSTM(
      (dropout): Dropout(p=0.0)
      (layers): ModuleList(
        (0): LSTMCell(640, 512)
      )
    )
    (attn): GlobalAttention(
      (linear_context): Linear(in_features=512, out_features=512, bias=False)
      (linear_query): Linear(in_features=512, out_features=512, bias=True)
      (v): Linear(in_features=512, out_features=1, bias=False)
      (linear_out): Linear(in_features=1024, out_features=512, bias=True)
      (softmax): Softmax()
      (tanh): Tanh()
    )
  )
  (generator): CopyGenerator(
    (linear): Linear(in_features=512, out_features=50004, bias=True)
    (linear_copy): Linear(in_features=512, out_features=1, bias=True)
    (softmax): Softmax()
    (sigmoid): Sigmoid()
  )
)
[2018-07-11 06:57:45,215 INFO] encoder: 7322624
[2018-07-11 06:57:45,215 INFO] encoder: 7322624
[2018-07-11 06:57:45,215 INFO] decoder: 35466581
[2018-07-11 06:57:45,215 INFO] decoder: 35466581
[2018-07-11 06:57:45,215 INFO] * number of parameters: 42789205
[2018-07-11 06:57:45,215 INFO] * number of parameters: 42789205
[2018-07-11 06:57:45,219 INFO] Start training...
[2018-07-11 06:57:45,219 INFO] Start training...
[2018-07-11 06:58:31,024 INFO] Loading train dataset from data/cnndm/CNNDM.train.1.pt, number of examples: 126095
[2018-07-11 06:58:31,024 INFO] Loading train dataset from data/cnndm/CNNDM.train.1.pt, number of examples: 126095
[2018-07-11 06:59:38,632 INFO] Loading train dataset from data/cnndm/CNNDM.train.1.pt, number of examples: 126095
[2018-07-11 06:59:38,632 INFO] Loading train dataset from data/cnndm/CNNDM.train.1.pt, number of examples: 126095
```

## Logs for working commit
Training continues through all epochs and model checkpoints are saved.
```
* number of parameters: 42789205
encoder:  7322624
decoder:  35466581
Making optimizer for training.

Start training...
 * number of epochs: 20, starting from Epoch 1
 * batch size: 16
Epoch  1,    50/ 7881; acc:   4.81; ppl: 860.66; xent:   6.76; 6527 src tok/s; 975 tgt tok/s;     47 s elapsed
Epoch  1,   100/ 7881; acc:  10.06; ppl: 379.20; xent:   5.94; 6780 src tok/s; 971 tgt tok/s;     92 s elapsed
Epoch  1,   150/ 7881; acc:  16.65; ppl: 259.27; xent:   5.56; 6801 src tok/s; 975 tgt tok/s;    137 s elapsed
Epoch  1,   200/ 7881; acc:  17.30; ppl: 251.41; xent:   5.53; 6361 src tok/s; 957 tgt tok/s;    185 s elapsed
Epoch  1,   250/ 7881; acc:  17.91; ppl: 227.53; xent:   5.43; 6621 src tok/s; 942 tgt tok/s;    231 s elapsed
Epoch  1,   300/ 7881; acc:  17.92; ppl: 238.38; xent:   5.47; 6573 src tok/s; 975 tgt tok/s;    278 s elapsed
Epoch  1,   350/ 7881; acc:  17.83; ppl: 253.62; xent:   5.54; 6478 src tok/s; 983 tgt tok/s;    326 s elapsed
Epoch  1,   400/ 7881; acc:  18.05; ppl: 213.80; xent:   5.37; 6726 src tok/s; 962 tgt tok/s;    371 s elapsed
Epoch  1,   450/ 7881; acc:  18.07; ppl: 213.01; xent:   5.36; 6697 src tok/s; 958 tgt tok/s;    416 s elapsed
Epoch  1,   500/ 7881; acc:  18.20; ppl: 217.72; xent:   5.38; 6520 src tok/s; 975 tgt tok/s;    464 s elapsed
Epoch  1,   550/ 7881; acc:  18.56; ppl: 188.91; xent:   5.24; 6521 src tok/s; 958 tgt tok/s;    510 s elapsed
```

**Comment by rbommasanimozilla** [comment]
I am currently training for summarization using master. One thing I noticed is you need to re-preprocess for the current master. 
This is what I am running (removed source and target directories):
python preprocess.py -src_seq_length 10000 -tgt_seq_length 10000 -src_seq_length_trunc 400 -tgt_seq_length_trunc 100 -dynamic_dict -share_vocab -max_shard_size 10485760
python train.py -gpuid 0 -copy_attn -global_attention mlp -word_vec_size 128 -rnn_size 256 -layers 1 -encoder_type brnn -train_steps 500000 -max_grad_norm 2 -dropout 0. -batch_size 16 -optim adagrad -learning_rate 0.15 -adagrad_accumulator_init 0.1 -reuse_copy_attn -copy_loss_by_seqlength -bridge -seed 777
